cleanlab/multiannotator.py

```diff
@@ -0,0 +1,467 @@
+    if labels_multiannotator.apply(
+        lambda s: np.array_equal(s.dropna().unique(), s.dropna()), axis=1
+    ).all():
+        warnings.warn("Annotators do not agree on any example. Check input data.")
+
+    # Compute the label quality scores for each annotators' labels
+    label_quality_scores_multiannotator = labels_multiannotator.apply(
+        _get_label_quality_scores_with_NA, args=[pred_probs], **kwargs
+    )
+    label_quality_scores_multiannotator = label_quality_scores_multiannotator.add_prefix(
+        "quality_annotator_"
+    )
+
+    # Count number of non-NaN values for each example
+    num_annotations = labels_multiannotator.count(axis=1).to_numpy()
+    label_quality_scores_multiannotator["num_annotations"] = num_annotations
+
+    # Compute consensus labels, annotator agreement and quality of consensus
+    consensus_label, annotator_agreement, quality_of_consensus = _get_consensus_stats(
+        labels_multiannotator=labels_multiannotator,
+        pred_probs=pred_probs,
+        num_annotations=num_annotations,
+        consensus_method=consensus_method[0]
+        if isinstance(consensus_method, list)
+        else consensus_method,
+        quality_method=quality_method,
+        **kwargs,
+    )
+
+    (
+        label_quality_scores_multiannotator["annotator_agreement"],
+        label_quality_scores_multiannotator["consensus_label"],
+        label_quality_scores_multiannotator["quality_of_consensus"],
+    ) = (
+        annotator_agreement,
+        consensus_label,
+        quality_of_consensus,
+    )
+
+    # compute consensus stats for alternative consensus methods (if provided)
+    if isinstance(consensus_method, list) and len(consensus_method) > 1:
+        for alt_consensus_method in consensus_method[1:]:
+            consensus_label_alt, _, quality_of_consensus_alt, = _get_consensus_stats(
+                labels_multiannotator=labels_multiannotator,
+                pred_probs=pred_probs,
+                num_annotations=num_annotations,
+                consensus_method=alt_consensus_method,
+                quality_method=quality_method,
+                **kwargs,
+            )
+
+            (
+                label_quality_scores_multiannotator[f"consensus_label_{alt_consensus_method}"],
+                label_quality_scores_multiannotator[f"quality_of_consensus_{alt_consensus_method}"],
+            ) = (
+                consensus_label_alt,
+                quality_of_consensus_alt,
+            )
+
+    # reordering the columns (return aggregated statistics first)
+    lqsm_columns = label_quality_scores_multiannotator.columns.tolist()
+    new_index = (
+        lqsm_columns[labels_multiannotator.shape[1] :]
+        + lqsm_columns[: labels_multiannotator.shape[1]]
+    )
+    label_quality_scores_multiannotator = label_quality_scores_multiannotator.reindex(
+        columns=new_index
+    )
+
+    annotator_stats = get_multiannotator_stats(labels_multiannotator, pred_probs)
```

Comment:
should only be computed if we need it (ie. if verbose or if return_annotator_stats)